### ISSUE TITLE
Multiple entity managers

### ISSUE THREAD

**Issue by silvandre95** [SymfonyCasts/reset-password-bundle#342]
Hello, I'm currently working with multiple entity managers and when I extend EntityRepository on my ResetPasswordRequestRepository, I get this error:

![image](https://github.com/user-attachments/assets/1e5f2773-87d0-4242-aca1-437d73698c23)

I know that when extending from EntityRepository we cannot rely on autowire, is this bundle prepared to work with multiple entity managers?

**Comment by bocharsky-bw** [comment]
I personally have never tried this bundle with multiple entity managers, but I think it should work well. The error is from your code, right? I mean, this `ResetPasswordRequestRepository` file is inside the src/ folder where you can modify it as you wish. Just take a look at the working repository class in that `App/Repository/Tenant/` folder, most probably you need to tweak that constructor method.

In short, just compare ResetPasswordRequestRepository's `construct()` with any other working repo `construct()`, I suppose they are different and you need to do some changes.

**Comment by silvandre95** [comment]
It is exactly like the other repositories, this is the only one I get the error. This is how I have the repo.

![image](https://github.com/user-attachments/assets/5aabef20-7806-43d6-9b48-7aa9d50f8dc7)

I believe the issue lies on the autowire of the ResetPasswordRequestRepositoryInterface on some files like ResetPasswordHelper.php. According to the multiple entity manager documentation, when we extend EntityRepository, we should not use autowire. But I need to extend EntityRepository for all, otherwise I can't work with multiple tenants.

**Comment by bocharsky-bw** [comment]
> According to the multiple entity manager documentation, when we extend EntityRepository, we should not use autowire

If so, then it's a problem because as I see the repo is injected in `ResetPasswordHelper` at least. Will the entity manager injection instead of the repo directly fix the problem? Do you see any other possible (better) solutions for this?

**Comment by silvandre95** [comment]
Yes, it can be a solution, but it has to be configurable, we have to be able to configure the entity manager we want otherwise it will always get the default one.

**Comment by bocharsky-bw** [comment]
Thanks for confirming. In this case, it seems this bundle is not ready to work with multiple entity managers. If you have any ideas on how to fix it - please, feel free to create a PR

**Comment by kbond** [comment]
I'll look into this. Should be possible but the bundle could be coded in a way that doesn't allow it. Basically, it should be using the manager registry to fetch repositories by entity.

**Comment by silvandre95** [comment]
@kbond if I extend the ServiceEntityRepository, I'll never be able to use this repo with my multi tenant implementation.

![image](https://github.com/user-attachments/assets/2be18e97-e0ad-4fbe-9ff6-e0c0198fd692)

**Comment by kbond** [comment]
Ahh, I misunderstood your multi-entity manager usage.

I think what you'll need to do is create your own `ResetPasswordRequestRepositoryInterface` service (that doesn't extend EntityRepository`) and implement it for use with your multi-tenant application logic.
